client/controllers/MenuController.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { useEffect, useRef, useState } from "react";
+import { CSSTransition, SwitchTransition } from "react-transition-group";
+import styled from "styled-components";
+
+import { CANVAS_SIZE } from "../../configs/canvas";
+import { EVENTS } from "../../configs/events";
+import { GAME_MODE, LOBBY_MODE, MENU_STATE } from "../../configs/game";
+import * as S from "../styled";
+import { LobbyContext, MenuContext } from "../utils/context";
+import { socket } from "../utils/socketSetup";
+import Death from "../view/menu/Death";
+import { Details } from "../view/menu/details";
+import MainMenu from "../view/menu/MainMenu";
+import Lobby from "./LobbyController";
+
+const MenuStyled = styled.div`
+  position: absolute;
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%, -50%) scale(1.5);
+  width: ${CANVAS_SIZE.WIDTH}px;
+  height: ${CANVAS_SIZE.HEIGHT}px;
+  opacity: 0.5;
+  z-index: 1;
+  overflow: hidden;
+`;
+
+const Title = styled.h1`
+  position: fixed;
+  font-size: 2em;
+  left: 50%;
+  transform: translateX(-50%);
+`;
+
+const MenuController = (): JSX.Element => {
```

Comment:
This class is quite big and does many things. It would be good to split some of its functionalities into different classes or methods (or at least parts). It's not necessary, but it would make it easier to read and refactor